imports.py

```diff
@@ -0,0 +1,203 @@
+	# if not decode_response_status_not_ok(response):
+	# 	response_dict = json.loads(response.text)
+	if 'entry' in response:
+		print("title: " + response['entry'][0]['resource']['title'])
+		print("resourceType: ", response['entry'][0]['resource']['resourceType'], end="  ")
+		print("id: " + response['entry'][0]['resource']['id'])
+		print("url: " + response['entry'][0]['resource']['url'])
+		print("full url: " + response['entry'][0]['fullUrl'])
+		id_idx = 1 if len(response['entry']) > 1 else 0
+		id_sys = response['entry'][id_idx]['resource']['identifier'][0]['system']
+		id_sys_val =  response['entry'][id_idx]['resource']['identifier'][0]['value']
+		print("identifier system: \"" + id_sys, end="\",  ")
+		print("value: \"" + id_sys_val + "\"")
+	else:
+		print("Not Present")
+
+
+def decode_validate_response(response, optional_prints=False):
+	"""Decode validate response"""
+	print(response)
+	if optional_prints:
+		print(response.keys())
+		print(response['issue']['severity'])
+		print(response['issue']['response'])
+		display(HTML(response['text']['div']))
+		print("")
+
+
+def decode_lookup_response(response: Dict):
+	"""Deocde lookup response"""
+	# if not decode_response_status_not_ok(response):
+	# 	response_dict = json.loads(response.text)
+	if 'issue' in response:
+		print("error")
+	else:
+		for p in response['parameter']:
+			if p['name'] != 'property' and p['name'] != 'designation':
+				if 'valueString' in p:
+					print(p['name'], ":", p['valueString'], sep="", end=", ")
+				if 'valueBoolean' in p:
+					print(p['name'], ":", p['valueBoolean'], sep="", end=", ")
+		print("")
+
+
+class TimsClient:
+	"""TIMS Client"""
+	http_client = httpx.Client()
+
+	def __init__(self, base_url: str, use_cache=False):
+		self.base_url = base_url
+		self.use_cache = use_cache
+
+	def do_hapi_request(self, internal_path: str, in_params: Dict = None) -> Dict:
+		"""GET request"""
+		url = os.path.join(self.base_url, internal_path)
+		if self.use_cache:
+			response_dict = cache_read(url, in_params)
+			if response_dict:
+				return response_dict
+		rq = httpx.Request("GET", url, params=in_params, headers=REQUEST_HEADERS)
+		response: Response = TimsClient.http_client.send(rq)
+		response_dict: Dict = json.loads(response.text)
+
+		cache_write(url, in_params, response_dict)
+		return response_dict
+
+	def summarize_code_systems(self, verbose=False):
+		"""Summarize code systems"""
+		response_dict = self.do_hapi_request("CodeSystem?_summary=true", {})
+		# if not decode_response_status_ok(response):
+		# 	response_dict = json.loads(response.text)
+		if verbose:
+			for vocab_entry in response_dict['entry']:
+				print(vocab_entry['resource']['name'])
+				print(vocab_entry['resource']['id'])
+				print(vocab_entry['fullUrl'])
+				print(vocab_entry['resource']['url'])
+				print('')
+		else:
+			sys_names = [x['resource']['name'] for x in response_dict['entry']]
+			print(', '.join(sys_names))
+
+	def summarize_code_system_by_name(self, name: str):
```

Comment:
I followed this basic pattern for each piece of functionality we want to demonstrate:
1. Should just need one method call for each functionality (after instantiating the client at the beginning).
2. Should have as few params as necessary.
3. Calls your existing `decode_*()` funcs.

Example:
```python
	def summarize_code_system_by_name(self, name: str):
		"""Summarize a code system by name param"""
		response: Dict = self.do_hapi_request("CodeSystem", {'name': name})
		# if not decode_response_status_not_ok(response):
		decode_code_system_json(response)
```

Reply:
I see the presentation looks much better for Chris and Melissa. I hoped we could find a way that leaves the url and query string in the output. That was why I chose httpx over requests.

Reply:
@chrisroederucdenver Oh, well we can do that even without `httpx`. Maybe they would prefer to see the URL (and params, if they appear outside the URL; though in most cases they are URL query params anyway). I could always have another flag at the top of the file for `VERBOSE` or `SHOW_URL`, or something like that. What do you think?

This would be an easy change for me to do so I can do this before Friday if you want.

Reply:
I can handle a fix like that. ;-)
I don't think they'll be looking at it super closely or critically, so we can roll with it and keep an eye on our commitments. The URL thing was for the dual purpose of testing, not just the presentation.